has been failing testing in datalad core for abit

from a sample run https://github.com/datalad/datalad/actions/runs/4270688403/jobs/7434642131

```
=========================== short test summary info ============================
FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - ValueError: unsupported URL 'garbledhttps://httpbin.org/status/404'
FAILED annexremotes/tests/test_uncurl.py::test_uncurl_addurl_unredirected - datalad.runner.exception.CommandError: CommandError: 'git -c diff.ignoreSubmodules=none annex addurl --file=dummy garbledhttps://httpbin.org/bytes/24 -c annex.dotfiles=true' failed with exitcode 1 under /tmp/pytest-of-runner/pytest-0/test_uncurl_addurl_unredirecte0/ds [out: 'addurl garbledhttps://httpbin.org/bytes/24 
failed'] [err: 'Configuration of annex.security.allowed-url-schemes does not allow accessing garbledhttps://httpbin.org/bytes/24
addurl: 1 failed']
====== 2 failed, 204 passed, 10 skipped, 12 warnings in 492.97s (0:08:12) ======
```

<details>
<summary>Seems started to fail on Feb 23rd</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2023/02[master]git
$> GIT_TERMINAL_PROMPT=0 datalad foreach-dataset --o-s relpath -r -J10 'git grep "annexremotes/tests/test_uncurl.py::test_uncurl_checkurl"' | grep PASS
07/pr/7273/5852580/github-Extensions-6132-success/6_test (datalad-next).txt:2023-02-07T03:45:45.3941139Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/7273/5852580/github-Extensions-6132-success/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-07T03:45:45.3940721Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/7273/5b40500/github-Extensions-6134-success/6_test (datalad-next).txt:2023-02-07T04:07:46.3473560Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/7273/5b40500/github-Extensions-6134-success/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-07T04:07:46.3473203Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/7280/9d0607f/github-Extensions-6131-success/6_test (datalad-next).txt:2023-02-07T03:37:00.8808077Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/7280/9d0607f/github-Extensions-6131-success/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-07T03:37:00.8807572Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/UNK/54b2009/github-Extensions-6130-success/6_test (datalad-next).txt:2023-02-07T03:30:53.2093499Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/pr/UNK/54b2009/github-Extensions-6130-success/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-07T03:30:53.2093140Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/push/maint/451d402/github-Extensions-6133-success/6_test (datalad-next).txt:2023-02-07T03:59:00.7867688Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
07/push/maint/451d402/github-Extensions-6133-success/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-07T03:59:00.7867294Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
17/pr/7289/6674407/github-Extensions-6135-failed/6_test (datalad-next).txt:2023-02-17T18:34:20.3762825Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
17/pr/7289/6674407/github-Extensions-6135-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-17T18:34:20.3762372Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7271/2340b4c/github-Extensions-6140-failed/6_test (datalad-next).txt:2023-02-21T20:43:52.9272196Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7271/2340b4c/github-Extensions-6140-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-21T20:43:52.9271818Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7292/a6d04eb/github-Extensions-6138-failed/6_test (datalad-next).txt:2023-02-21T17:51:54.5743063Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7292/a6d04eb/github-Extensions-6138-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-21T17:51:54.5742680Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7292/db062bb/github-Extensions-6139-failed/6_test (datalad-next).txt:2023-02-21T17:52:30.0783368Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7292/db062bb/github-Extensions-6139-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-21T17:52:30.0782885Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7293/5983e07/github-Extensions-6143-failed/6_test (datalad-next).txt:2023-02-21T21:00:05.3347304Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/pr/7293/5983e07/github-Extensions-6143-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-21T21:00:05.3346843Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/push/maint/3cca0db/github-Extensions-6141-failed/6_test (datalad-next).txt:2023-02-21T20:50:20.3239495Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/push/maint/3cca0db/github-Extensions-6141-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-21T20:50:20.3238901Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/push/maint/6c2a573/github-Extensions-6142-failed/6_test (datalad-next).txt:2023-02-21T20:53:58.8893521Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
21/push/maint/6c2a573/github-Extensions-6142-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-21T20:53:58.8893136Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
22/push/maint/589dd65/github-Extensions-6144-failed/6_test (datalad-next).txt:2023-02-22T14:17:14.2291479Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
22/push/maint/589dd65/github-Extensions-6144-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-22T14:17:14.2290836Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
23/pr/7298/cb86f33/github-Extensions-6145-failed/6_test (datalad-next).txt:2023-02-23T08:21:03.3728395Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
23/pr/7298/cb86f33/github-Extensions-6145-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-23T08:21:03.3728159Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
23/pr/7301/f884340/github-Extensions-6147-failed/6_test (datalad-next).txt:2023-02-23T13:40:54.7131815Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
23/pr/7301/f884340/github-Extensions-6147-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-23T13:40:54.7131442Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl PASSED
```
```
$> GIT_TERMINAL_PROMPT=0 datalad foreach-dataset --o-s relpath -r -J10 'git grep "annexremotes/tests/test_uncurl.py::test_uncurl_checkurl"' | grep FAIL
23/pr/7301/95fd21c/github-Extensions-6146-failed/6_test (datalad-next).txt:2023-02-23T13:37:13.0638685Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl FAILED
23/pr/7301/95fd21c/github-Extensions-6146-failed/6_test (datalad-next).txt:2023-02-23T13:50:18.1610897Z FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - AssertionError: assert False
23/pr/7301/95fd21c/github-Extensions-6146-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-23T13:37:13.0638162Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl FAILED
23/pr/7301/95fd21c/github-Extensions-6146-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-23T13:50:18.1610893Z FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - AssertionError: assert False
24/pr/7303/9f6003a/github-Extensions-6148-failed/6_test (datalad-next).txt:2023-02-24T03:38:23.8123241Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl FAILED
24/pr/7303/9f6003a/github-Extensions-6148-failed/6_test (datalad-next).txt:2023-02-24T03:49:08.6291997Z FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - ValueError: unsupported URL 'garbledhttps://httpbin.org/status/404'
24/pr/7303/9f6003a/github-Extensions-6148-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-24T03:38:23.8122755Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl FAILED
24/pr/7303/9f6003a/github-Extensions-6148-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-24T03:49:08.6291991Z FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - ValueError: unsupported URL 'garbledhttps://httpbin.org/status/404'
25/pr/7305/8a145bf/github-Extensions-6149-failed/6_test (datalad-next).txt:2023-02-25T02:27:15.6902746Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl FAILED
25/pr/7305/8a145bf/github-Extensions-6149-failed/6_test (datalad-next).txt:2023-02-25T02:35:10.6044091Z FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - ValueError: unsupported URL 'garbledhttps://httpbin.org/status/404'
25/pr/7305/8a145bf/github-Extensions-6149-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-25T02:27:15.6902293Z annexremotes/tests/test_uncurl.py::test_uncurl_checkurl FAILED
25/pr/7305/8a145bf/github-Extensions-6149-failed/test (datalad-next)/13_datalad-next tests using pytest.txt:2023-02-25T02:35:10.6044073Z FAILED annexremotes/tests/test_uncurl.py::test_uncurl_checkurl - ValueError: unsupported URL 'garbledhttps://httpbin.org/status/404'
```
</details>

--- a/datalad_next/annexremotes/tests/test_uncurl.py
+++ b/datalad_next/annexremotes/tests/test_uncurl.py
@@ -145,7 +145,7 @@ def test_uncurl_checkurl(tmp_path):
     # of external redirector service
     r.url_tmpl = f'{hbsurl}/redirect-to?url={{origurl}}'
     r.match = [
-        re.compile('.*(?P<origurl>http://.*)$'),
+        re.compile('.*(?P<origurl>https?://.*)$'),
     ]
     assert not r.checkurl(f'garbled{hbsurl}/status/404')
     assert r.checkurl(f'garbled{hbsurl}/bytes/24')
@@ -157,7 +157,7 @@ def test_uncurl_addurl_unredirected(existing_dataset):
     dsca = ds.repo.call_annex
     # same set as in `test_uncurl_checkurl()`
     dsca(['initremote', 'myuncurl'] + std_initargs + [
-        'match=.*(?P<origurl>http://.*)$',
+        'match=.*(?P<origurl>https?://.*)$',
         f'url={hbsurl}/redirect-to?url={{origurl}}',
     ])
     # feed it a broken URL, which must be getting fixed by the rewritting

